subsys/net/l2/ethernet/arp.c

```diff
@@ -535,11 +535,15 @@ static inline struct net_pkt *arp_prepare_reply(struct net_if *iface,
 
 static bool arp_hdr_check(struct net_arp_hdr *arp_hdr)
 {
+	struct in_addr src_ip;
+
+	memcpy(&src_ip, arp_hdr->src_ipaddr, sizeof(src_ip));
```

Comment:
Do we really need intermediate variable here, can't we just do
```
net_ipv4_is_addr_loopback(UNLIGNED_GET(...));
```
here?

Reply:
I don't think we could use `UNALIGNED_GET()` in this case, the function expects a pointer, but this macro returns a value, and it's not possible to obtain the address of the result, it ends up in a compilation error:
```
error: lvalue required as unary '&' operand
```
I tried however a different approach - instead of using the local copy, simply feed the function with the raw buffer address, typecasted into `in_addr *` in this case - the compiler had no issue with that. All these little address check functions use `UNALIGNED_GET()` macro internally to access the structure fields anyway, so I think it should be safe to provide it even with an unaligned address, but maybe I miss something, I don't know.

But yeah, now that all of the "hotspots" are identified, we can think of possible optimizations.

Reply:
I would say that we try to not do both, which is clearly overkill:
- if we use a buffer variable, let's not use `GET_UNALIGNED` on it
- if we use the raw packed variable, let's use `GET_UNALIGNED` on it